### PR TITLE
feat(layout): sidebar accordion

### LINKED ISSUE #51: Avatars get updated on presence update
Most notably, when this happens, avatars flicker (presumably because they reload). This is corroborated by `UPDATE_AVATAR` being committed to the store.

### PR DESCRIPTION
**Pull request by misterupkeep**
An accordion on the sidebar now resides, splitting in half, vertically, into `Roster` and `Recents`. The contents of both are the same, and are to be changed in further commits.

This branch also fixes #51, as well as an issue with the `<SelfBar/>` component displaying the wrong data (see: 14993ee)

### CODE REVIEW

**Comment by misterupkeep** [comment]
@00drip If you decide to merge, do a **merge commit**. The two `fix` commits are out of place and I didn't feel like moving them onto another branch. Very unorthodox and unprofessional;

I'm somewhat worried about 4fe30c6, so if we ever need to revert it, we don't have to lose the accordion.

File: components/Accordion/Item.vue [hunk 1]
```diff
@@ -0,0 +1,56 @@
+<template>
+  <div class="d-flex flex-column accordion-item" :class="{'flex-grow-1': expanded}">
```
Comment (00drip) [inline components/Accordion/Item.vue:2]: Do you think we should eventually consider the Vue dynamic class or style syntax an antipattern? We could probably move this kind of thing into computed props eventually, it'll make the longer but more clear.

File: components/SideBar.vue [hunk 2]
```diff
@@ -1,21 +1,51 @@
...
   export default {
+    components: {AccordionItem},
+    data() {
+      return {
+        expanded: 0
+      }
+    },
+    methods: {
+      toggle() {
+        this.expanded = (this.expanded + 1) % 2
```
Comment (00drip) [inline components/SideBar.vue:46]: Are you joking

File: components/SideBar.vue [hunk 3]
```diff
@@ -1,21 +1,51 @@
...
-    <simplebar class="narrow-scrollbar flex-grow-1">
-      <roster-list
-          :pinned="[]"
-          :items="items"
-          :selected-jid="selectedJid"/>
-    </simplebar>
+    <header-bar/>
+    <div class="d-flex flex-grow-1 flex-column">
+        <accordion-item header="Roster"
+                        :expanded="!expanded"
```
Comment (00drip) [inline components/SideBar.vue:6]: There's probably a better way about this than a bunch of NOTs. I mean we should just number the things and check against the number, if we ever end up with more accordion items.

File: store/index.js [hunk 4]
```diff
@@ -253,10 +260,19 @@ export const actions = {
         }))]);
     },
 
-    async [$actions.downloadAvatar]({ commit, state }, { jid }) {
+    /**
```
Comment (00drip) [inline store/index.js:263]: Good stuff documenting everything. I think we should arrange the actual store variables/getters/setters better and maybe add some comments to categorize them better, it's a bit of a mess as is.

**Review by 00drip — COMMENTED** [review]
Good to go. Nothing worth changing right now.

File: components/Accordion/Item.vue (hunk 1 above)
Comment (misterupkeep) [reply components/Accordion/Item.vue:2]: I think it's fine for one- at most two- classes. It gets out of hand when it's multiple lines, and especially when there's a mix of `[]` and `{}` class syntax (like in some components). For just one class, the benefits are questionable.

File: components/SideBar.vue (hunk 2 above)
Comment (misterupkeep) [reply components/SideBar.vue:46]: I think this was written in mind for three accordion ribs, but then I never added the third one, and I realised toggling could be replaced with inline !nots.

File: components/SideBar.vue (hunk 3 above)
Comment (misterupkeep) [reply components/SideBar.vue:6]: Ideally, the items would be generated with a `<template v-for="(rib, i) in ribs"/>` and dynamic components. Then the number thing could work, i.e. `:expanded="expanded === i"`

**Pull request merged**